rosdistro_reviewer/element_analyzer/rosdep.py

```diff
@@ -0,0 +1,356 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+# Licensed under the Apache License, Version 2.0
+
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from colcon_core.logging import colcon_logger
+from colcon_core.plugin_system import satisfies_version
+from rosdep2 import create_default_installer_context
+from rosdistro_reviewer.element_analyzer \
+    import ElementAnalyzerExtensionPoint
+from rosdistro_reviewer.review import Annotation
+from rosdistro_reviewer.review import Criterion
+from rosdistro_reviewer.review import Recommendation
+from rosdistro_reviewer.yaml_changes import get_changed_yaml
+from rosdistro_reviewer.yaml_changes import prune_changed_yaml
+
+logger = colcon_logger.getChild(__name__)
+
+EOL_PLATFORMS = {
+    'debian': {
+        'lenny',
+        'squeeze',
+        'wheezy',
+        'jessie',
+        'stretch',
+        'buster',
+    },
+    'fedora': {
+        str(n) for n in range(21, 39)
+    },
+    'rhel': {
+        str(n) for n in range(3, 8)
+    },
+    'ubuntu': {
+        'trusty',
+        'utopic',
+        'vivid',
+        'wily',
+        'xenial',
+        'yakkety',
+        'zesty',
+        'artful',
+        'bionic',
+        'cosmic',
+        'disco',
+        'eoan',
+        'groovy',
+        'hirsute',
+        'impish',
+        'kinetic',
+        'lunar',
+        'mantic',
+    },
+}
+
+
+def _check_key_names(criteria, annotations, changed_rosdeps, key_counts):
+    # Bypass check if no new keys were added
+    if not any(
+        getattr(key, '__lines__', None)
+        for changes in changed_rosdeps.values()
+        for key in changes.keys()
+    ):
+        return
+
+    recommendation = Recommendation.APPROVE
+    problems = set()
+
+    # Pip-only rules should end in -pip
+    for file, changes in changed_rosdeps.items():
+        if Path(file).name != 'python.yaml':
+            continue
```

Comment:
~~Is there another check that asserts that python changes are in python.yaml? The use of this file is convention only so this could miss pip rules that "dodge" by adding to base.yaml.~~

Found lines 95-109.